mvn-extension/src/test/java/org/arquillian/smart/testing/mvn/ext/SurefireReportStorageTest.java

```diff
@@ -0,0 +1,110 @@
+        surefireReportsDir = folder.newFolder(SUREFIRE_REPORTS_DIR_NAME);
+        projectDir = folder.newFolder();
+
+        project = mock(Model.class);
+        when(project.getProjectDirectory()).thenReturn(projectDir);
+
+        Build build = mock(Build.class);
+        when(project.getBuild()).thenReturn(build);
+        when(build.getDirectory()).thenReturn(surefireReportsDir.getParent());
+    }
+
+    @Test
+    public void should_copy_xml_files() throws IOException {
+        // given
+        Map<String, Long> expectedReports = feedWithReports(surefireReportsDir);
+
+        // when
+        SurefireReportStorage.copySurefireReports(project);
+
+        // then
+        File reportsDir = new File(projectDir, TEMP_REPORT_DIR);
+        assertThat(reportsDir).exists();
+
+        Map<String, Long> actualReports = Arrays.stream(reportsDir.listFiles())
+            .collect(Collectors.toMap(File::getName, File::length));
+        assertThat(actualReports).containsAllEntriesOf(expectedReports).hasSameSizeAs(expectedReports);
+    }
+
+    @Test
+    public void should_purge_reports_directory() throws IOException {
+        // given
+        SurefireReportStorage.copySurefireReports(project);
+        MavenSession mavenSession = mock(MavenSession.class);
+        MavenProject mavenProject = mock(MavenProject.class);
+        when(mavenProject.getModel()).thenReturn(project);
+        when(mavenSession.getAllProjects()).thenReturn(Arrays.asList(new MavenProject[] {mavenProject}));
+        File reportsDir = new File(projectDir, TEMP_REPORT_DIR);
+
+        // when
+        assertThat(reportsDir).exists();
+        SurefireReportStorage.purgeReports(mavenSession);
+
+        // then
+        assertThat(reportsDir).doesNotExist();
+    }
+
+    @Test
+    public void should_not_create_nor_copy_anything() throws IOException {
+        // given
+        surefireReportsDir.delete();
+
+        // when
+        new SurefireReportStorage().copySurefireReports(project);
+
+        // then
+        File reportsDir = new File(projectDir, TEMP_REPORT_DIR);
+        assertThat(reportsDir).doesNotExist();
+    }
+
+    private Map<String, Long> feedWithReports(File surefireReportsDir) throws IOException {
+        Map<String, Long> expectedReports = new HashMap<>();
+        expectedReports.put("first-report.xml", createDummyFile(surefireReportsDir, "first-report.xml"));
+        expectedReports.put("second-report.xml", createDummyFile(surefireReportsDir, "second-report.xml"));
+        expectedReports.put("third-report.xml", createDummyFile(surefireReportsDir, "third-report.xml"));
+        createDummyFile(surefireReportsDir, "not-xml-report");
+        return expectedReports;
+    }
+
+    private long createDummyFile(File directory, String fileName) throws IOException {
+        File file = new File(directory, fileName);
+        RandomAccessFile raf = new RandomAccessFile(file, "rw");
+        raf.setLength(fileName.length());
```

Comment:
Why do we need to `setLength` here?

Reply:
What would you suggest @dipak-pawar ?

Reply:
I am thinking appending/adding some text instead of setting length for a file which doesn't have any content inside.

Reply:
I'm just wondering why we need RAF at all here?

Reply:
The idea is creating dummy but not empty file by using RAF and setting the length of the file. Yeah, I could put the string there, that is another option. From my point of view, this seemed to be faster, simpler and easier, that's why I used this approach...

Reply:
Fair enough, at the end of the day, it's testing dummy so we can live with that :)